gateway/ln-gateway/src/actor.rs

```diff
@@ -38,6 +39,7 @@ pub struct GatewayActor {
     task_group: TaskGroup,
     gw_rpc: GatewayRpcSender,
     pub sender: Sender<Arc<AtomicBool>>,
+    registration: LightningGateway,
```

Comment:
I know you're just refactoring this but its a bit confusing to me. Why does the "actor" need a handle to the `LightningGateway`? Actors ideally should just be federation clients. Is this needed for the federation public key?